app/jobs/fetch_constituencies_job.rb

```diff
@@ -29,6 +29,8 @@ def perform
           end
 
           constituency.save!
+
+          update_constituencies_parliaments(constituency)
```

Comment:
I think we need to do this outside the constituencies loop and do it for each parliament, e.g.

``` ruby
constituencies.each do |external_id, name, ons_code, start_date, end_date|
# existing code
end

parliaments.each do |parliament|
  parliament.constituency_ids = Constituency.for_parliament(parliament).external_ids
end

class Constituency < ApplicationRecord
  class << self
    def for_parliament(parliament)
      if parliament.archived? || parliament.dissolved?
        between(parliament.opening_at, parliament.dissolution_at)
      else
        current
      end
    end

    def between(opening_at, dissolution_at)
      where(arel_table[:start_date].lteq(opening_at).and(arel_table[:end_date].gteq(dissolution_at).or(arel_table[:end_date].eq(nil))))
    end

    def external_ids
      pluck(:external_id)
    end
  end
end
```

We'll need to add some tests for these scopes - check the following:

1. Parliaments that have been archived
2. Parliaments that have been dissolved
3. Parliaments that are current before opening
4. Parliaments that are current after opening

For each of those scenarios, check that

1. Constituencies ended before parliament opened are excluded
2. Constituencies started after parliament opened are excluded
3. Constituencies started before parliament opened and ended after dissolution are included
4. Constituencies started before parliament opened and not ended are included